grammar.js

```diff
@@ -1,284 +1,391 @@
+module.exports = grammar({
+  name: 'openscad',
 
-const rules = {
-  source_file: $ => repeat(choice($.use_statement, $._item)),
+  word: ($) => $.identifier,
 
-  // http://stackoverflow.com/questions/13014947/regex-to-match-a-c-style-multiline-comment/36328890#36328890
-  comment: $ => token(choice(
-    seq('//', /(\\(.|\r?\n)|[^\\\n])*/),
-    seq(
-      '/*',
-      /[^*]*\*+([^/*][^*]*\*+)*/,
-      '/'
-    )
-  )),
+  supertypes: ($) => [$._literal, $._expression],
 
-  _item: $ => choice(
-    seq($.assignment, ';'),
-    $._statement,
-    $.module_declaration,
-    $.function_declaration,
-  ),
+  extras: ($) => [
+    /\s|\\\r?\n/,
+    $.comment,
+  ],
+
+  rules: {
+    source_file: ($) => repeat(choice($.use_statement, $._item)),
 
-  // modules
-  module_declaration: $ => seq(
-    'module',
-    field('name', $.identifier),
-    field('parameters', $.parameters_declaration),
-    field('body', $._statement),
-  ),
-  parameters_declaration: $ => parens(seq(commaSep($._parameter_declaration), optional(','))),
-  _parameter_declaration: $ => choice($._variable_name, $.assignment),
+    // http://stackoverflow.com/questions/13014947/regex-to-match-a-c-style-multiline-comment/36328890#36328890
+    comment: () => token(choice(
+      seq('//', /(\\(.|\r?\n)|[^\\\n])*/),
+      seq(
+        '/*',
+        /[^*]*\*+([^/*][^*]*\*+)*/,
+        '/',
+      ),
+    )),
 
-  // function declarations are slightly different from $.function, which is for
-  // function literals
-  function_declaration: $ => seq(
-    'function',
-    field('name', $.identifier),
-    field('parameters', $.parameters_declaration),
-    '=', $._expression
-  ),
+    _item: ($) => choice(
+      seq($.assignment, ';'),
+      $._statement,
+      $.module_declaration,
+      $.function_declaration,
+    ),
 
-  // statements are language constructs that can create objects
-  _statement: $ => choice(
-    $.for_block,
-    $.intersection_for_block,
-    $.if_block,
-    $.let_block,
-    $.assign_block,
-    $.union_block,
-    $.modifier_chain,
-    $.transform_chain,
-    $.include_statement,
-    $.assert_statement,
-    ';',
-  ),
-  // use/include statements
-  // These are called statements, but use statements aren't included in
-  // $._statement because they can't be used in all the same places as other
-  // statements
-  include_statement: $ => seq('include', $.include_path),
-  use_statement: $ => seq('use', $.include_path),
-  include_path: $ => /<[^>]*>/,
-  assignment: $ => seq(
-    field('left', $._variable_name),
-    '=',
-    field('right', $._expression)
-  ),
-  union_block: $ => curlies(repeat($._item)),
+    // modules
+    module_declaration: ($) => seq(
+      'module',
+      field('name', $.identifier),
+      field('parameters', $.parameters_declaration),
+      field('body', $._statement),
+    ),
+    parameters_declaration: ($) => parens(seq(commaSep($._parameter_declaration), optional(','))),
```

Comment:
This `commaSep` invocation should probably now use the optional `trailing_separator` argument.

Reply:
Fails with tests since (,) is valid